Source/WebKit/UIProcess/ios/WKContentView.mm

```diff
@@ -815,25 +821,46 @@ - (void)_didRelaunchProcess
     [self _accessibilityRegisterUIProcessTokens];
     [self setUpInteraction];
 #if HAVE(VISIBILITY_PROPAGATION_VIEW)
-    [self _setupVisibilityPropagationViewForWebProcess];
+    [self _setupVisibilityPropagationForWebProcess];
 #if ENABLE(GPU_PROCESS)
-    [self _setupVisibilityPropagationViewForGPUProcess];
+    [self _setupVisibilityPropagationForGPUProcess];
 #endif
 #endif
 }
 
 #if HAVE(VISIBILITY_PROPAGATION_VIEW)
 - (void)_webProcessDidCreateContextForVisibilityPropagation
 {
-    [self _setupVisibilityPropagationViewForWebProcess];
+    [self _setupVisibilityPropagationForWebProcess];
 }
 
 - (void)_gpuProcessDidCreateContextForVisibilityPropagation
 {
-    [self _setupVisibilityPropagationViewForGPUProcess];
+    [self _setupVisibilityPropagationForGPUProcess];
 }
+
+#if USE(EXTENSIONKIT)
+- (WKVisibilityPropagationView *)_createVisibilityPropagationView
+{
+    if (!WebKit::AuxiliaryProcessProxy::manageProcessesAsExtensions())
+        return nil;
+
+    if (!_visibilityPropagationViews)
+        _visibilityPropagationViews = [NSHashTable weakObjectsHashTable];
+
+    RetainPtr visibilityPropagationView = adoptNS([[WKVisibilityPropagationView alloc] init]);
+    [_visibilityPropagationViews addObject:visibilityPropagationView.get()];
+
+    [self _setupVisibilityPropagationForWebProcess];
+#if ENABLE(GPU_PROCESS)
+    [self _setupVisibilityPropagationForGPUProcess];
 #endif
 
+    return visibilityPropagationView.autorelease();
```

Comment:
Will the reference count of visibilityPropagationView be zero when the RetainPtr is deallocated upon returning from this function?

Reply:
No, `.autorelease()` prevents that.